src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy

```diff
@@ -533,6 +551,26 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
             }
             
             buildTarget.finalizedBy tasks.removeImage
+
+            ${imageIdValidation()}
+        """
+    }
+
+    private static String imageIdValidation() {
+        """
+            tasks.withType(DockerBuildImage).all { Task task ->
+                def assertTask = tasks.create("assertImageIdFor\${task.name}"){
+                    def dependantTask = task
+                    doLast {
+                        def value = dependantTask.getImageId().getOrNull()
+                        if(value == null || !(value ==~ /^\\w+\$/)) {
```

Comment:
Just as a side note - not required to change. It's probably good enough to use `assert` instead of having to throw a `GradleException`.